controllers/api.template.js

```diff
@@ -22,7 +24,23 @@ router.get('/tpl/:id', async (req, res, next) => {
       return;
     }
 
-   // Increment template downloads
+    // Check no referrer
+    if(!req.headers.referer){        
+        res.writeHead(302, {
+          'Location': '/template/'+template.id+'/'+template.slug
+        });     
+        res.end();  
+    }      
+    // Check no-allowed-referrer
+    const referer_hostname= new URL(req.headers.referer).hostname;      
+    if(allowed_referrers.indexOf(referer_hostname)===-1){           
+        res.writeHead(302, {
+          'Location': '/template/'+template.id+'/'+template.slug
+        });     
```

Comment:
1. Make the new location a `const` declared at the top of the router so it can be easily changed if necessary
2. Combine these two into the same `if` block - now there's unnecessary repetition of the exact same code.
3. Use `res.redirect()` instead of `.writeHead()`? The former is an express.js feature, which automatically sets status to 302 and handles the redirect with a nice response.

Reply:
Fixed